lib/nerves_hub_link/script.ex

```diff
@@ -0,0 +1,46 @@
+defmodule NervesHubLink.Script do
+  use GenServer
+
+  @doc """
+  Run a script from NervesHub and capture its output
+  """
+  def capture(text, ref) do
+    _ = GenServer.start(__MODULE__, {self(), text, ref})
```

Comment:
Shouldn't this be `start_link`'d so that if the process calling `capture/2`, the code doesn't continue to run? I also wonder if there should be a limit to the number of scripts being run simultaneously.

Reply:
No, I don’t think so at least. Since we didn’t want the script to kill the socket we don’t want to link it during capture

Reply:
If the script crashes, an error should be returned to the server. If the connection crashes for other reasons, then is there any reason for the script to run since it can't return results to the server any more?

Reply:
I’m going to assume I misunderstood what Jon wanted since what you described is how it worked before

Reply:
Yeah, nothing can be easy, can it? I took Jon's comment

It feels like this may be a temporary fix to an immediately pressing issue, and we'll be revisiting it soon. In case you haven't seen other threads, there's #220 which is discussing an extension system. This would make it easier to add non-firmware update features without worrying about the possible ramifications to the core functionality.

I do think that we should return errors when things go wrong to make issues easier to debug. I read Jon's comment as don't block the main connection process since it would block processing of firmware update requests.

Reply:
It seems like we can wait for async feedback over the next day, and then decide what's most practical to do for #221 tomorrow.